quickwit/quickwit-actors/src/mailbox.rs

```diff
@@ -157,13 +159,49 @@ impl<A: Actor> Mailbox<A> {
         &self,
         message: M,
     ) -> Result<oneshot::Receiver<A::Reply>, SendError>
+    where
+        A: Handler<M>,
+        M: 'static + Send + Sync + fmt::Debug,
+    {
+        self.send_message_with_backpressure_counter(message, None)
+            .await
+    }
+
+    /// Sends a message to the actor owning the associated inbox.
+    ///
+    /// If the actor experiences some backpressure, then
+    /// `backpressure_microsecs` will be increased by the amount of
+    /// microseconds of backpressure experienced.
+    pub async fn send_message_with_backpressure_counter<M>(
+        &self,
+        message: M,
+        backpressure_microsecs_opt: Option<&IntCounter>,
+    ) -> Result<oneshot::Receiver<A::Reply>, SendError>
     where
         A: Handler<M>,
         M: 'static + Send + Sync + fmt::Debug,
     {
         let (envelope, response_rx) = wrap_in_envelope(message);
-        self.inner.tx.send_low_priority(envelope).await?;
-        Ok(response_rx)
+        if let Some(backpressure_microsecs) = backpressure_microsecs_opt {
+            match self.inner.tx.try_send_low_priority(envelope) {
```

Comment:
A couple of notes here.

1- We use try_send to make the cost is minimal if we do not have any backpressure. 
2- We do not use `is_full()` because, `is_full() is true` even without backpressure, if the queue capacity is 0.